docs/distribution/launcher.md

```diff
@@ -0,0 +1,208 @@
+---
+layout: developer-doc
+title: Enso Launcher
+category: distribution
+tags: [distribution, launcher]
+order: 4
+---
+
+# Enso Launcher
+The launcher is used to run Enso commands (like the REPL, language server etc.)
+and seamlessly manage Enso versions. This document describes it's features. Its
+command-line interface is described in the [CLI](./launcher-cli.md) document.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Launcher Distribution](#launcher-distribution)
+- [Project Management](#project-management)
+  - [Creating a Project](#creating-a-project)
+  - [Per-Project Enso Version](#per-project-enso-version)
+  - [Project Configuration](#project-configuration)
+- [Enso and Graal Version Management](#enso-and-graal-version-management)
+  - [GraalVM Override](#graalvm-override)
+  - [Downloading Enso Releases](#downloading-enso-releases)
+  - [Downloading GraalVM Releases](#downloading-graalvm-releases)
+- [Running Enso Components](#running-enso-components)
+- [Global User Configuration](#global-user-configuration)
+- [Updating the Launcher](#updating-the-launcher)
+  - [Minimal Required Launcher Version](#minimal-required-launcher-version)
+  - [Downloading Launcher Releases](#downloading-launcher-releases)
+
+<!-- /MarkdownTOC -->
+
+## Launcher Distribution
+The launcher is distributed as a native binary for each platform (Windows,
+Linux, Mac). It is distributed in a ZIP archive as described in
+[Enso Home Layout](./distribution.md#enso-home-layout), except that the
```

Comment:
I think we should get the newest Enso already archived with the launcher in the default bundle (while having optional launcher-only bundles available). Lowers the initial annoyance factor.

Reply:
I'm all for this.

Reply:
Should this be the case for just the releases or for all published artifacts on the CI?

Reply:
So each release should contain the following packages?
- `enso-bundle-Windows-<version>.zip`
- `enso-bundle-Linux-<version>.zip`
- `enso-bundle-macOS-<version>.zip`
- `enso-engine-<version>.zip`
- `enso-launcher-only-Windows-<version>.zip`
- `enso-launcher-only-Linux-<version>.zip`
- `enso-launcher-only-macOS-<version>.zip`

Reply:
Yep, but don't call it `enso-launcher-only`, please.

Reply:
and we should specify architecture with OS.